src/calendar/model/CalendarModel.ts

```diff
@@ -149,232 +90,231 @@ export class CalendarModelImpl implements CalendarModel {
 			!repeatRulesEqual(newEvent.repeatRule, existingEvent.repeatRule)
 		) {
 			// We should reload the instance here because session key and permissions are updated when we recreate event.
-			return this._doCreate(newEvent, zone, groupRoot, newAlarms, existingEvent).then(() =>
-				this._entityClient.load<CalendarEvent>(CalendarEventTypeRef, newEvent._id),
-			)
+			await this.doCreate(newEvent, zone, groupRoot, newAlarms, existingEvent)
+			return await this.entityClient.load<CalendarEvent>(CalendarEventTypeRef, newEvent._id)
 		} else {
 			newEvent._ownerGroup = groupRoot._id
 			// We can't load updated event here because cache is not updated yet. We also shouldn't need to load it, we have the latest
 			// version
-			return this._calendarFacade.updateCalendarEvent(newEvent, newAlarms, existingEvent).then(() => newEvent)
+			await this.calendarFacade.updateCalendarEvent(newEvent, newAlarms, existingEvent)
+			return newEvent
 		}
 	}
 
-	loadCalendarInfos(progressMonitor: IProgressMonitor): Promise<Map<Id, CalendarInfo>> {
-		const user = this._logins.getUserController().user
+	/** Load map from group/groupRoot ID to the calendar info */
+	async loadCalendarInfos(progressMonitor: IProgressMonitor): Promise<Map<Id, CalendarInfo>> {
+		const user = this.logins.getUserController().user
 
 		const calendarMemberships = user.memberships.filter((m) => m.groupType === GroupType.Calendar)
 		const notFoundMemberships: GroupMembership[] = []
-		return promiseMap(calendarMemberships, (membership) =>
-			Promise.all([
-				this._entityClient.load(CalendarGroupRootTypeRef, membership.group).then((it) => {
-					progressMonitor.workDone(1)
```

Comment:
what happened to this `workDone()`?

Reply:
all three of them were merged and moved to line 126 (along with the one in the catch block). we're loading three instances in parallel and it's not too useful to make the progress that granular.

Reply:
but i just noticed that line 126 should probably be inside the catch block